lib/model/eval.js

```diff
@@ -30,12 +44,17 @@ const socketEventHandler = (event, socket, data, callback) => {
       response = {
         name: 'invalid',
       };
+      logger.moduleLog('debug', 'Eval Model', data);
+      callback(response);
+      socket.close();
       break;
     case 'submission_pending':
       logger.moduleLog('debug', 'Eval Model', 'Submission Pending...');
       response = {
         name: 'submission_pending',
       };
+      logger.moduleLog('debug', 'Eval Model', data);
```

Comment:
The log messages for all the cases are same. They need to be different.

Reply:
Sir, all of them change depending on the data they receive. Below is an example log :
<details>
<pre>
2018-12-09 08:31:02 [debug] EVAL INPUT: Fetching id to submit.
2018-12-09 08:31:02 [debug] EVAL: Evaluate request for
2018-12-09 08:31:02 [debug] EVAL: {"lab":"Lab1","lang":"java","commitHash":"","idNo":"************"}
2018-12-09 08:31:02 [debug] EVAL MODEL: Opening a socket connection to http://autolab.bits-goa.ac.in:9000
2018-12-09 08:31:02 [debug] EVAL MODEL:
    course details
    {"name":"Object Oriented Programming","course number":"CS F213","instructor in charge":"Dr. Rahul Thakur","other instructors":["Prof. Neena Goveas"]}
2018-12-09 08:31:02 [debug] EVAL MODEL:
    labs_status
    {"Lab_No":{"Lab_No":"Lab0","start_date":"13","start_month":"08","start_year":"2018","start_hour":"16","start_minute":"00","end_date":"14","end_month":"08","end_year":"2018","end_hour":"14","end_minute":"00","hard_date":"14","hard_month":"08","hard_year":"2018","hard_hour":"14","hard_minute":"00","penalty":"0"},"status":0}
    {"Lab_No":{"Lab_No":"Lab1","start_date":"14","start_month":"08","start_year":"2018","start_hour":"15","start_minute":"00","end_date":"14","end_month":"08","end_year":"2018","end_hour":"17","end_minute":"00","hard_date":"14","hard_month":"08","hard_year":"2018","hard_hour":"17","hard_minute":"00","penalty":"0"},"status":0}
    {"Lab_No":{"Lab_No":"Lab2","start_date":"21","start_month":"08","start_year":"2018","start_hour":"15","start_minute":"00","end_date":"21","end_month":"08","end_year":"2018","end_hour":"17","end_minute":"00","hard_date":"21","hard_month":"08","hard_year":"2018","hard_hour":"17","hard_minute":"00","penalty":"0"},"status":0}
    {"Lab_No":{"Lab_No":"Lab3","start_date":"28","start_month":"08","start_year":"2018","start_hour":"15","start_minute":"00","end_date":"28","end_month":"08","end_year":"2018","end_hour":"17","end_minute":"00","hard_date":"28","hard_month":"08","hard_year":"2018","hard_hour":"17","hard_minute":"00","penalty":"0"},"status":0}
    {"Lab_No":{"Lab_No":"Lab4","start_date":"04","start_month":"09","start_year":"2018","start_hour":"15","start_minute":"00","end_date":"04","end_month":"09","end_year":"2018","end_hour":"18","end_minute":"05","hard_date":"04","hard_month":"09","hard_year":"2018","hard_hour":"18","hard_minute":"05","penalty":"0"},"status":0}
    {"Lab_No":{"Lab_No":"Lab5","start_date":"11","start_month":"09","start_year":"2018","start_hour":"15","start_minute":"00","end_date":"11","end_month":"09","end_year":"2018","end_hour":"17","end_minute":"00","hard_date":"11","hard_month":"09","hard_year":"2018","hard_hour":"17","hard_minute":"00","penalty":"0"},"status":0}
    {"Lab_No":{"Lab_No":"Lab6Prelab","start_date":"23","start_month":"09","start_year":"2018","start_hour":"00","start_minute":"00","end_date":"25","end_month":"09","end_year":"2018","end_hour":"12","end_minute":"00","hard_date":"25","hard_month":"09","hard_year":"2018","hard_hour":"12","hard_minute":"00","penalty":"0"},"status":0}
    {"Lab_No":{"Lab_No":"Lab6","start_date":"25","start_month":"09","start_year":"2018","start_hour":"15","start_minute":"00","end_date":"25","end_month":"09","end_year":"2018","end_hour":"17","end_minute":"15","hard_date":"25","hard_month":"09","hard_year":"2018","hard_hour":"17","hard_minute":"15","penalty":"0"},"status":0}
    {"Lab_No":{"Lab_No":"Lab7","start_date":"16","start_month":"10","start_year":"2018","start_hour":"15","start_minute":"05","end_date":"16","end_month":"10","end_year":"2018","end_hour":"17","end_minute":"00","hard_date":"16","hard_month":"10","hard_year":"2018","hard_hour":"17","hard_minute":"00","penalty":"0"},"status":0}
    {"Lab_No":{"Lab_No":"Lab8","start_date":"30","start_month":"10","start_year":"2018","start_hour":"15","start_minute":"05","end_date":"30","end_month":"10","end_year":"2018","end_hour":"17","end_minute":"00","hard_date":"30","hard_month":"10","hard_year":"2018","hard_hour":"17","hard_minute":"00","penalty":"0"},"status":0}
    {"Lab_No":{"Lab_No":"Lab9","start_date":"13","start_month":"11","start_year":"2018","start_hour":"15","start_minute":"05","end_date":"13","end_month":"11","end_year":"2018","end_hour":"21","end_minute":"00","hard_date":"13","hard_month":"11","hard_year":"2018","hard_hour":"21","hard_minute":"00","penalty":"0"},"status":0}
    {"Lab_No":{"Lab_No":"Lab10","start_date":"20","start_month":"11","start_year":"2018","start_hour":"15","start_minute":"05","end_date":"20","end_month":"11","end_year":"2018","end_hour":"21","end_minute":"00","hard_date":"20","hard_month":"11","hard_year":"2018","hard_hour":"21","hard_minute":"00","penalty":"0"},"status":0}
2018-12-09 08:31:02 [debug] EVAL MODEL:
    lab_data
    {"course":"{\n    \"name\": \"Object Oriented Programming\",\n    \"course number\": \"CS F213\",\n    \"instructor in charge\": \"Dr. Rahul Thakur\",\n    \"other
instructors\": [\n        \"Prof. Neena Goveas\"\n    ]\n}","lab":"{\n    \"Labs\": [\n        {\n            \"Lab_No\": \"Lab0\",\n            \"start_date\": \"13\",\n            \"start_month\": \"08\",\n            \"start_year\": \"2018\",\n            \"start_hour\": \"16\",\n            \"start_minute\": \"00\",\n            \"end_date\": \"14\",\n            \"end_month\": \"08\",\n            \"end_year\": \"2018\",\n            \"end_hour\": \"14\",\n            \"end_minute\": \"00\",\n
           \"hard_date\": \"14\",\n            \"hard_month\": \"08\",\n            \"hard_year\": \"2018\",\n            \"hard_hour\": \"14\",\n            \"hard_minute\": \"00\",\n            \"penalty\": \"0\"\n        },\n        {\n            \"Lab_No\": \"Lab1\",\n            \"start_date\": \"14\",\n            \"start_month\": \"08\",\n            \"start_year\": \"2018\",\n            \"start_hour\": \"15\",\n            \"start_minute\": \"00\",\n            \"end_date\": \"14\",\n
       \"end_month\": \"08\",\n            \"end_year\": \"2018\",\n            \"end_hour\": \"17\",\n            \"end_minute\": \"00\",\n            \"hard_date\": \"14\",\n            \"hard_month\": \"08\",\n            \"hard_year\": \"2018\",\n            \"hard_hour\": \"17\",\n            \"hard_minute\": \"00\",\n
 \"penalty\": \"0\"\n        },\n        {\n            \"Lab_No\": \"Lab2\",\n            \"start_date\": \"21\",\n            \"start_month\": \"08\",\n            \"start_year\": \"2018\",\n            \"start_hour\": \"15\",\n            \"start_minute\": \"00\",\n            \"end_date\": \"21\",\n            \"end_month\": \"08\",\n            \"end_year\": \"2018\",\n            \"end_hour\": \"17\",\n            \"end_minute\": \"00\",\n            \"hard_date\": \"21\",\n            \"hard_month\": \"08\",\n            \"hard_year\": \"2018\",\n            \"hard_hour\": \"17\",\n            \"hard_minute\": \"00\",\n            \"penalty\": \"0\"\n
  },\n        {\n            \"Lab_No\": \"Lab3\",\n            \"start_date\": \"28\",\n            \"start_month\": \"08\",\n            \"start_year\": \"2018\",\n
          \"start_hour\": \"15\",\n            \"start_minute\": \"00\",\n            \"end_date\": \"28\",\n            \"end_month\": \"08\",\n            \"end_year\": \"2018\",\n            \"end_hour\": \"17\",\n            \"end_minute\": \"00\",\n            \"hard_date\": \"28\",\n            \"hard_month\": \"08\",\n
   \"hard_year\": \"2018\",\n            \"hard_hour\": \"17\",\n            \"hard_minute\": \"00\",\n            \"penalty\": \"0\"\n        },\n        {\n
  \"Lab_No\": \"Lab4\",\n            \"start_date\": \"04\",\n            \"start_month\": \"09\",\n            \"start_year\": \"2018\",\n            \"start_hour\": \"15\",\n            \"start_minute\": \"00\",\n            \"end_date\": \"04\",\n            \"end_month\": \"09\",\n            \"end_year\": \"2018\",\n            \"end_hour\": \"18\",\n            \"end_minute\": \"05\",\n            \"hard_date\": \"04\",\n            \"hard_month\": \"09\",\n            \"hard_year\": \"2018\",\n            \"hard_hour\": \"18\",\n            \"hard_minute\": \"05\",\n            \"penalty\": \"0\"\n        },\n        {\n            \"Lab_No\": \"Lab5\",\n
          \"start_date\": \"11\",\n            \"start_month\": \"09\",\n            \"start_year\": \"2018\",\n            \"start_hour\": \"15\",\n            \"start_minute\": \"00\",\n            \"end_date\": \"11\",\n            \"end_month\": \"09\",\n            \"end_year\": \"2018\",\n            \"end_hour\": \"17\",\n
       \"end_minute\": \"00\",\n            \"hard_date\": \"11\",\n            \"hard_month\": \"09\",\n            \"hard_year\": \"2018\",\n            \"hard_hour\": \"17\",\n            \"hard_minute\": \"00\",\n            \"penalty\": \"0\"\n        },\n        {\n            \"Lab_No\": \"Lab6Prelab\",\n            \"start_date\": \"23\",\n            \"start_month\": \"09\",\n            \"start_year\": \"2018\",\n            \"start_hour\": \"00\",\n            \"start_minute\": \"00\",\n
           \"end_date\": \"25\",\n            \"end_month\": \"09\",\n            \"end_year\": \"2018\",\n            \"end_hour\": \"12\",\n            \"end_minute\": \"00\",\n            \"hard_date\": \"25\",\n            \"hard_month\": \"09\",\n            \"hard_year\": \"2018\",\n            \"hard_hour\": \"12\",\n
  \"hard_minute\": \"00\",\n            \"penalty\": \"0\"\n        },\n        {\n            \"Lab_No\": \"Lab6\",\n            \"start_date\": \"25\",\n            \"start_month\": \"09\",\n            \"start_year\": \"2018\",\n            \"start_hour\": \"15\",\n            \"start_minute\": \"00\",\n            \"end_date\": \"25\",\n            \"end_month\": \"09\",\n            \"end_year\": \"2018\",\n            \"end_hour\": \"17\",\n            \"end_minute\": \"15\",\n            \"hard_date\": \"25\",\n            \"hard_month\": \"09\",\n            \"hard_year\": \"2018\",\n            \"hard_hour\": \"17\",\n            \"hard_minute\": \"15\",\n            \"penalty\": \"0\"\n        },\n        {\n            \"Lab_No\": \"Lab7\",\n            \"start_date\": \"16\",\n            \"start_month\": \"10\",\n
          \"start_year\": \"2018\",\n            \"start_hour\": \"15\",\n            \"start_minute\": \"05\",\n            \"end_date\": \"16\",\n            \"end_month\": \"10\",\n            \"end_year\": \"2018\",\n            \"end_hour\": \"17\",\n            \"end_minute\": \"00\",\n            \"hard_date\": \"16\",\n
     \"hard_month\": \"10\",\n            \"hard_year\": \"2018\",\n            \"hard_hour\": \"17\",\n            \"hard_minute\": \"00\",\n            \"penalty\": \"0\"\n        },\n        {\n            \"Lab_No\": \"Lab8\",\n            \"start_date\": \"30\",\n            \"start_month\": \"10\",\n            \"start_year\": \"2018\",\n            \"start_hour\": \"15\",\n            \"start_minute\": \"05\",\n            \"end_date\": \"30\",\n            \"end_month\": \"10\",\n
 \"end_year\": \"2018\",\n            \"end_hour\": \"17\",\n            \"end_minute\": \"00\",\n            \"hard_date\": \"30\",\n            \"hard_month\": \"10\",\n            \"hard_year\": \"2018\",\n            \"hard_hour\": \"17\",\n            \"hard_minute\": \"00\",\n            \"penalty\": \"0\"\n        },\n        {\n            \"Lab_No\": \"Lab9\",\n            \"start_date\": \"13\",\n            \"start_month\": \"11\",\n            \"start_year\": \"2018\",\n            \"start_hour\": \"15\",\n            \"start_minute\": \"05\",\n            \"end_date\": \"13\",\n            \"end_month\": \"11\",\n            \"end_year\": \"2018\",\n
           \"end_hour\": \"21\",\n            \"end_minute\": \"00\",\n            \"hard_date\": \"13\",\n            \"hard_month\": \"11\",\n            \"hard_year\": \"2018\",\n            \"hard_hour\": \"21\",\n            \"hard_minute\": \"00\",\n            \"penalty\": \"0\"\n        },\n        {\n            \"Lab_No\": \"Lab10\",\n            \"start_date\": \"20\",\n            \"start_month\": \"11\",\n            \"start_year\": \"2018\",\n            \"start_hour\": \"15\",\n
      \"start_minute\": \"05\",\n            \"end_date\": \"20\",\n            \"end_month\": \"11\",\n            \"end_year\": \"2018\",\n            \"end_hour\": \"21\",\n            \"end_minute\": \"00\",\n            \"hard_date\": \"20\",\n            \"hard_month\": \"11\",\n            \"hard_year\": \"2018\",\n
\"hard_hour\": \"21\",\n            \"hard_minute\": \"00\",\n            \"penalty\": \"0\"\n        }\n    ]\n}"}
2018-12-09 08:31:12 [debug] EVAL MODEL: Sucessfully Sumbitted. Evaluated results
2018-12-09 08:31:12 [debug] EVAL MODEL: {"marks":["1","1","1","1","1","1","1","1","1","1"],"comment":["Accepted","Accepted","Accepted","Accepted","Accepted","Accepted","Accepted","Accepted","Accepted","Accepted"],"id_no":"************","Lab_No":"Lab1","commit":"","time":"2018-12-09 03:03:45","status":0,"penalty":0,"socket":"PV-SViPDcqGSQoZkAAsX","log":"<removed>"}
2018-12-09 08:31:12 [debug] EVAL OUTPUT: Evaluation logs: NO LOGS FOR THIS EVALUATION.
</pre>
</details>